docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+which consists of three types of metadata documents: an entry point metadata
+document (``zarr.json``), array metadata documents, and group metadata
+documents. Each type of metadata document is described in the following
+subsections.
+
+Metadata documents are defined here using the JSON
+type system defined in [RFC8259]_. In this section, the terms "value",
+"number", "string" and "object" are used to denote the types as
+defined in [RFC8259]_. The term "array" is also used as defined in
+[RFC8259]_, except where qualified as "Zarr array". Following
+[RFC8259]_, this section also describes an object as a set of
+name/value pairs. This section also defines how metadata documents are
+encoded for storage.
+
+
+Only the top level metadata document ``zarr.json`` is guaranteed to be of JSON
+type, and can be used to define other formats for array-level and group-level
+metadata documents. In the case where non-JSON metadata documents are used in a
+Zarr hierarchy, the following sections on group and array level metadata are
+non-normative, but other metadata formats are expected to define some
+equivalence relations with the JSON documents.
+
+
+Entry point metadata
+--------------------
+
+Each Zarr hierarchy must have an entry point metadata document, which
+provides essential information regarding the format version being
+used, the encoding being used for group and array metadata, and any
+extensions that affect the layout or interpretation of data
+in the store.
+
+The entry point metadata document must contain a single object
+containing the following names:
+
+``zarr_format``
+^^^^^^^^^^^^^^^
+
+    A string containing the URI of the Zarr core
+    specification that defines the metadata format. For Zarr
+    hierarchies conforming to this specification, the value must be
+    the string "https://purl.org/zarr/spec/core/3.0".
+
+    Implementations of this specification may assume that the final path
+    segment of this URI ("3.0") represents the core specification version
+    number, where "3" is the major version number and "0" is the minor
+    version number. Implementations of this specification may also assume
+    that future versions of this specification that retain the same major
+    versioning number ("3") will be backwards-compatible, in the sense
+    that any new features added to the specification can be safely
+    ignored. In other words, if the major version number is "3",
+    implementations of this specification may read and interpret metadata
+    as defined in this specification, ignoring any name/value pairs
+    where the name is not defined here. See also the `stability policy`_.
+
+    Note that this value is given as a URI rather than as a simple
+    version number string to help with discovery of this
+    specification.
+
+``metadata_encoding``
+^^^^^^^^^^^^^^^^^^^^^
+
+    A string containing the URI pointing to a document describing the method
+    used for encoding group and array metadata documents.
```

Comment:
I don't think that we should specify the mimetype here, since that's not used in the core protocol, and it applies only for specific stores. Might be worth to note in the respective store specs, though.